ocaml/idl/datamodel_observer.ml

```diff
@@ -0,0 +1,157 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let register =
+  call ~name:"register" ~in_oss_since:None ~lifecycle:[] ~hide_from_docs:true
+    ~doc:"Register a new observer on all hosts"
```

Comment:
There's a parameter for selecting a single host, is the observer set up on all hosts?

Reply:
register has now been moved to host level, this is called by create and load (also host level). It acts as the function responsible for calling the library. Create is its parent function that sets it up on other hosts

Reply:
Then teh doc needs to be changed to reflect that

Reply:
I see that i will update the doc string

Reply:
Fixed this